app/src/main/java/org/openobservatory/ooniprobe/common/worker/UpdateDescriptorsWorker.kt

```diff
@@ -0,0 +1,173 @@
+                        descriptor.runId,
+                        applicationContext
+                    )
+
+                if (descriptor.shouldUpdate(updatedDescriptor)) {
+                    updatedDescriptor.isAutoUpdate = descriptor.isAutoUpdate
+                    updatedDescriptor.isAutoRun = descriptor.isAutoRun
+
+                    Log.d(TAG, "Saving updates for ${descriptor.runId}")
+
+                    updatedDescriptor.save()
+                    updatedDescriptors.add(updatedDescriptor)
+                }
+            }
+
+            val outputData = Data.Builder()
+                .putString(
+                    KEY_UPDATED_DESCRIPTORS,
+                    (applicationContext as Application).gson.toJson(updatedDescriptors)
+                ).build()
+
+            Log.e(TAG, "Descriptor updates complete")
+
+            Result.success(outputData)
+
+        } catch (exception: Exception) {
+            Log.e(TAG, "Error Updating")
+            exception.printStackTrace()
+            ThirdPartyServices.logException(exception)
+            Result.failure()
+        }
+    }
+
+    companion object {
+        @JvmField
+        var UPDATED_DESCRIPTORS_WORK_NAME =
+            "${AutoUpdateDescriptorsWorker::class.java.name}.UPDATED_DESCRIPTORS_WORK_NAME"
+
+        private val TAG = AutoUpdateDescriptorsWorker::class.java.simpleName
+
+        private val UPDATE_DESCRIPTOR_CHANNEL: String =
+            AutoUpdateDescriptorsWorker::class.java.simpleName
+
+        private val KEY_UPDATED_DESCRIPTORS =
+            "${AutoUpdateDescriptorsWorker::class.java.name}.KEY_UPDATED_DESCRIPTORS"
+    }
+}
+
+class ManualUpdateDescriptorsWorker(
+    context: Context,
+    workerParams: WorkerParameters
+) : Worker(context, workerParams) {
+
+    override fun doWork(): Result {
+
+        setProgressAsync(Data.Builder().putInt(PROGRESS, 0).build())
+
+        val app = applicationContext.applicationContext as Application
+        app.serviceComponent.inject(d)
+
+        return try {
+            Log.d(TAG, "Fetching descriptors from input")
+
+            val updatedDescriptors: ArrayList<TestDescriptor> = ArrayList()
+
+            val descriptors = inputData.getLongArray(KEY_DESCRIPTOR_IDS)?.let {
+                d.testDescriptorManager.getDescriptorsFromIds(it.toTypedArray())
+            }.run {
+                d.testDescriptorManager.getDescriptorWithAutoUpdateDisabled()
+            }
+
+            if(descriptors.isEmpty()) {
+                Log.e(TAG, "No descriptors to update")
+                return Result.success()
+            }
+
+            for (descriptor in descriptors) {
+                Log.d(TAG, "Fetching updates for ${descriptor.runId}")
+
+                val updatedDescriptor: TestDescriptor =
+                    d.testDescriptorManager.fetchDescriptorFromRunId(
+                        descriptor.runId,
+                        applicationContext
+                    )
+                /**
```

Comment:
AFAIK, JavaDoc comments only have effect when documenting classes and methods. Did I got this incorrectly? Is this comment rendered in the documentation? If not, maybe I'd avoid using JavaDoc inside methods.